### PR TITLE
Move back test configs

### PR DESCRIPTION
**Pull request by ronald-cron-arm**
## Description

Companion PR of #9771 

## PR checklist
- [x] **changelog** not required, just moving test config files 
- [x] **development PR** https://github.com/Mbed-TLS/mbedtls/pull/9771
- [x] **framework PR** https://github.com/Mbed-TLS/mbedtls-framework/pull/79
- [x] **3.6 PR** here
- [x] **2.28 PR** not required because: 4.0/3.6 only work
- **tests**  not required because: just moving test config files

### CODE REVIEW

**Comment by ronald-cron-arm** [comment]
On OpenCI only a glitch in an DTLS test thus the CI can be considered green.

**Review by Harry-Ramsey — APPROVED** [review]
LGTM.

**Review by minosgalanakis — APPROVED** [review]
LGTM

File: tests/configs/config_test_driver.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Mbed TLS configuration for PSA test driver libraries. It includes:
```
Comment (minosgalanakis) [inline tests/configs/config_test_driver.h:2]: I assume that moving forward there will be two versions of the test driver?

- `tests/configs/config_test_driver.h` for 3.6
- `tf-psa-crypto/tests/configs/config_test_driver.h` for devel, and 4.0 onwards? 

Would it be good to add a comment to link those, so when someone is adding a feature they will know about it to backport it here?

**Pull request merged**